xbanish.c

```diff
@@ -107,6 +109,10 @@ main(int argc, char *argv[])
 				usage(argv[0]);
 			}
 			break;
+		case 't':
+			timeout = strtoul(optarg, NULL, 0);
+			signal(SIGALRM, (void *)hide_cursor);
```

Comment:
You can't directly call X11 functions from a signal handler.  You'd probably need to use the `XSyncAlarm` stuff like [xdimmer](https://github.com/jcs/xdimmer/blob/master/xdimmer.c) does, or just poke a pipe and use poll (also like xdimmer does).

Reply:
by "You can't", what are you referring to? it certainly works fine, so is there some spec specifying you shouldnt?

Reply:
Yes, just Google for "xlib from signal handler".  From the [X FAQ](http://www.phy.ohio.edu/computer/xwin/xfaq.html):

> Subject: 150)  Can I make Xt or Xlib calls from a signal handler?
> 
> No. Xlib and Xt have no mutual exclusion for protecting critical sections. If
> your signal handler makes such a call at the wrong time (which might be while
> the function you are calling is already executing), it can leave the library
> in an inconsistent state.
 
It also describes a way to handle signals using pipe/poll as I mentioned.

Reply:
ah, I see. that might actually explain the mystery freezes I've had in the past. I'll look into it